### ISSUE TITLE
[Bug]:  Ctrl+C failed to shutdown DN with certain orders

### ISSUE THREAD

**Issue by gouhongshen** [matrixorigin/matrixone#10926]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): ff25bf2374c6c4a26d5dee23d4574eda38abfc82
- Hardware parameters: Mac
- OS type:
- Others:
```

### Actual Behavior

failed to shut down the DN service using `ctrl + c` (`SIGINT`) in the local environment (non-docker) when starting the cluster with the distributed mod (`-cfg`). it reports:

```
2023/07/31 11:06:19.031202 +0800 WARN stopper/stopper.go:209 tasks still running in stopper {"stopper": "main", "continuous": "30.005220834s", "tasks": "trace,dn-service"}

2023/07/31 11:06:19.840744 +0800 WARN dn-service stopper/stopper.go:209 tasks still running in stopper {"uuid": "dd4dccb4-4d3c-41f8-b482-5251dc7a41bf", "stopper": "txn rpc server", "continuous": "30.004839667s", "tasks": "undefined"}
```

### Expected Behavior

shutdown correctly.

### Steps to Reproduce

```Markdown
failed to shutdown(ctrl + c) DN  with shutdown orders:

multi-CN: ./mo-service -cfg etc/launch-tae-multi-CN-tae-DN/

1. log -> cn1 -> dn -> cn2
2. log -> cn1 -> cn2 -> dn
3. log -> dn -> cn1 -> cn2


signle cn: ./mo-service -cfg etc/launch-tae-CN-tae-DN/

1. log -> dn -> cn
2. log -> cn -> dn
```
```


### Additional information

_No response_

**Comment by gouhongshen** [comment]
related issue #10887

**Comment by gouhongshen** [comment]
"undefined" task is `github.com/matrixorigin/matrixone/pkg/txn/rpc.(*server).handleTxnRequest`, it will blocked by `txt.Wait()` when `commit1PC()`.

and `driverAppender::append()` also needs 180s to retry timeout.

**Comment by gouhongshen** [comment]
when ctrl+c shut down the log service, the dn service cannot append entries to the log service, after retrying 180s, the append task will panic. 

however, all append tasks are submitted to a goroutine pool to be scheduled, and the pool will catch the panic through `recover`, and do some operations specified by users. 

In the previous implementation, no `panic handler` is specified to the pool when calling the 'ants.NewPool`, so the `panic` can only crash the goroutine it belongs to instead of the dn service. 

after the append task crashes, some other tasks will fall into the dead wait, which expected the append task to return and do some further work.

that is why the ctrl+c cannot shut down the dn service.

the fix is to specify a panic handler.

**Comment by gouhongshen** [comment]
fixed by PR #10988

**Comment by gouhongshen** [comment]
PR #10988 fixed the problem of panic will be caught by `recover`, which makes the whole dn can panic when the append retry failed. 

and PR #11006 enables the notification when the dn is supposed to be shut down, by which the dn can panic timely when it is in a long append retry process.
